precip spatial cells and fix for sparklines sort order

dblodgett-usgs opened the pull request:
This outputs an RDS with grid cells we can use with the GDP. They've been uploaded to the GDP as in #40.

dblodgett-usgs committed:
incremental commit for some review

dblodgett-usgs committed:
incremental commit for some review

dblodgett-usgs committed:
Merge branch 'master' of https://github.com/USGS-VIZLAB/hurricane-irma

dblodgett-usgs committed:
rewrite. This is good now. ready to integrate into viz.

dblodgett-usgs committed:
rewrite. This is good now. ready to integrate into viz.

dblodgett-usgs committed:
Merge branch 'master' of https://github.com/USGS-VIZLAB/hurricane-irma

dblodgett-usgs committed:
rename

dblodgett-usgs committed:
rename

dblodgett-usgs committed:
Merge branch 'master' of https://github.com/USGS-VIZLAB/hurricane-irma

dblodgett-usgs committed:
reformat view-limits add bbox as output.

dblodgett-usgs committed:
reformat view-limits add bbox as output.

dblodgett-usgs committed:
precip spatial outputs cells as rds now.

dblodgett-usgs committed:
precip spatial outputs cells as rds now.

dblodgett-usgs committed:
Merge branch 'master' of https://github.com/USGS-VIZLAB/hurricane-irma

jordansread commented on scripts/fetch/fetch.precip_spatial.R:
```diff
@@ -0,0 +1,64 @@
+  
+  ### constants
+  cell_size <- .1 # degrees -- set to make grid more course or dense.
+  states <- c("florida", "georgia", "alabama", "south carolina") # from maps states id list.
```
to do this I think you want to depend on the other states that come in through fetch, otherwise, this isn't as re-usable.

dblodgett-usgs replied:
I use this for subsetting the ones that come in from the rest of the package. Is there a standard subset somewhere else in the package?

jordansread commented on scripts/fetch/fetch.precip_spatial.R:
```diff
@@ -0,0 +1,64 @@
+                                                                                stringsAsFactors = F, row.names = states))
+  state_boundary <- maptools::unionSpatialPolygons(state_boundary, state_boundary@data$viz)
+  
+  sp_cells@proj4string <- CRS("+init=epsg:4326")
```
can you use the shared EPSG code in `view-limits`?

dblodgett-usgs replied:
Figured I'd do everything in here as 4326 then project later. But happy to do whatever works.

dblodgett-usgs replied:
I'm switching over the the projection from the project.

jordansread commented on scripts/fetch/fetch.view_limits.R:
```diff
@@ -1,13 +1,20 @@
+                     width = viz[['width']], 
+                     height = viz[['height']], 
+                     pointsize = viz[['pointsize']],
+                     latlonbbox = bbox))
```
how does latlonbbox differ from the ylim and xlim in the transform? just the lat/lon version of those? or a different window?

dblodgett-usgs replied:
lat lon version, yeah.

dblodgett-usgs replied:
Will remove this. It'll be better to use the project projection. Only trick may be if the GDP doesn't like it.

dblodgett-usgs committed:
Updates per review using project proj now.

dblodgett-usgs committed:
merge

dblodgett-usgs commented:
My latest should #41 and includes commits that are from #44 to try and avoid conflicts.

dblodgett-usgs committed:
Fixes #40 getting precip for cell geometry.

dblodgett-usgs commented:
Just added another fetcher fashioned about the county precip GDP fetcher. It outputs a csv file keyed on the ids in the cells. I just added this in parallel to not break anything. Still need to switch out the counties, but I think we have all the data we need to do that now.

dblodgett-usgs committed:
Merge branch 'master' of https://github.com/USGS-VIZLAB/hurricane-irma

dblodgett-usgs committed:
Merge branch 'master' of github.com:jread-usgs/hurricane-irma

dblodgett-usgs committed:
Just some reformatting for readability

dblodgett-usgs committed:
Moved the sparkline sorting to the very end of the processing chain. Still something weird with spark line vertical sort.

dblodgett-usgs committed:
cleaning up a bit of the sites.sp stuff to use a more reliable row matching method.

dblodgett-usgs committed:
Merge branch 'master' of https://github.com/USGS-VIZLAB/hurricane-irma

dblodgett-usgs commented:
This PR is dumb... sorry.

I turned off fetch for my precip cell stuff that's still unmerged.

see: https://github.com/USGS-VIZLAB/hurricane-irma/pull/43/commits/3fee41e9a582b3dc5d8de82d0327ea9559c8da6c for the real fix for sorting. 

The other two commits I made recently are cleanup of some code that was either hard to read or super fragile / prone to losing track of sort order.
